sky/backends/backend_utils.py

```diff
@@ -1221,7 +1221,8 @@ def _get_tpu_vm_pod_ips(ray_config: Dict[str, Any],
     cluster_name = ray_config['cluster_name']
     zone = ray_config['provider']['availability_zone']
     query_cmd = (f'gcloud compute tpus tpu-vm list --filter='
-                 f'\\(labels.ray-cluster-name={cluster_name}\\) '
+                 f'"(labels.ray-cluster-name={cluster_name} AND '
+                 f'state!=PREEMPTED)" '
                  f'--zone={zone} --format=value\\(name\\)')
```

Comment:
I think this makes sense. 

A minor thing that may not be good to fix in this PR is that in L1150, we directly return the IPs, without checking whether the number of nodes in the cluster matches the expected nodes as we done in L1209. Can we raise the `exceptions.FetchIPError(.FetchIPError.Reason.HEAD)` instead of handling it separately below?
https://github.com/skypilot-org/skypilot/blob/2abcde326caff998be60b788276de25afaacb500/sky/backends/backend_utils.py#L1620-L1622

Also, `get_node_ips` also checks the ray cluster is correctly running on the cluster. Is it possible for TPU VM having multiple nodes? If yes, we may want to check the healthiness of the ray cluster as well.

Reply:
Ah yes, we should make the behavior consistent and a part of the issue is related to https://github.com/skypilot-org/skypilot/pull/1185.
I'll make sure that PR fixes it and merge it asap.